docs/developers/fuelmint.md

```diff
@@ -0,0 +1,163 @@
+# Fuelmint Documentation
+
+## Introduction
+
+### Fuel
+
+Fuel is an execution layer for the Modular stack.
+
+Fuel is an execution layer in modular blockchains
+that provides a fast and efficient solution
+to scalability. It aims to address the scalability
+crisis in Ethereum, which has seen modest increase
+in total throughput with L2 solutions. Fuel is designed
+to leverage the shift in L1 blockchain architectures
+from monolithic to modular, where execution is separated
+from data availability and consensus, leading to increased
+bandwidth capacity.
+
+Fuel has three key components:
+
+* Parallel Transaction Execution
+* The Fuel Virtual Machine
+* A great developer experience
+
+Fuel provides a flexible throughput with high security
+and is considered as the engine for the autonomous future,
+where individuals and groups can access verifiable
+autonomous systems.
+
+Learn about Fuel [here](https://www.fuel.network/)
+
+### Celestia
+
+Celestia is a modular blockchain network whose goal
+is to build a scalable data availability layer,
+enabling the next generation of scalable blockchain
+architectures - modular blockchains.
+
+You can learn more about Celestia in our
+[Concepts](../../concepts/how-celestia-works/introduction)
+and [Build Modular](../build-modular) pages.
+
+### Fuelmint
+
+Fuelmint is a Sovereign FuelVM rollups on Celestia using
+Rollmint and ABCI. You can read the announcement blog post
+[here](https://blog.celestia.org).
+
+In this tutorial, we will show you how to setup Fuelmint, so
+you can test out the advantages of using Sway while having a sovereign
+rollup on Celestia.
+
+Let's get started!
+
+## Installation
+
+### Dependencies
+
+You would need to install the following dependencies on your machine
+to be able to setup Fuelmint.
+
+* [Install Rust and Cargo](https://doc.rust-lang.org/cargo/getting-started/installation.html)
+* [Install Docker](https://docs.docker.com/engine/install/ubuntu/)
+* [Install System Environment Setup for Linux, including Golang](../../nodes/environment)
+
+### Setup Fuelmint
+
+Fuelmint Repo is [here](https://github.com/Ferret-san/fuelmint/tree/tx_pool_experiment)
+
+First step is to clone the Fuelmint repo:
+
+```bash
+git clone https://github.com/Ferret-san/fuelmint
+```
+
+Then go to the docker directory:
+
+```bash
+cd fuelmint/local-da
+```
+
+You can start by running a Docker Compose
+setup with a local DA network
+
+```bash
+docker compose -f ./docker/test-docker-compose.yml up
+```
+
+In a separate terminal session, you must
+build the Fuelmint binary with `cargo`:
+
+```bash
+cd ..
+cargo run --bin fuelmint
+```
+
+In another terminal session, you will need
+to build the Rollkit Node with Golang:
+
+```bash
+cd rollkit-node
+go mod edit -replace github.com/cosmos/cosmos-sdk=github.com/celestiaorg/cosmos-sdk-rollmint@v0.46.7-rollmint-v0.5.0-no-fraud-proofs
```

Comment:
hm, I updated the Rollkit command, but am getting this [error](https://app.warp.dev/block/2UWuKEZlOEBfMx46y1AxBm)

Reply:
that's because there's no makefile with an `install` instruction. This must have slipped in by accident. The actual steps are:
```
cd rollkit-node
go build
```

Reply:
and then you can run `./rollkit-node` with flags from the directory where the binary lives

Reply:
what directory does the binary end up in?

Reply:
also confirming: user does not need to install rollkit themselves?

Reply:
I suspect there may be some changes required here: https://github.com/Ferret-san/fuelmint/blob/7083e811cdc006576a5d08c41de7c9c1a5b762bb/rollkit-node/main.go#L14 and other places where `rollmint` is imported